src/v/model/record_batch_reader.cc

```diff
@@ -224,6 +224,8 @@ make_fragmented_memory_storage_batches(Container batches) {
             } else {
                 data.emplace_back(std::exchange(data_chunk, {}));
             }
+            data_chunk.reserve(
+              std::min(elements_per_fragment, i - batches.size()));
```

Comment:
isn't `i - batches.size()` always negative so it's close to size_t::max?

Reply:
Good catch!